src/main/java/com/esri/samples/map/generate_offline_map_with_local_basemap/GenerateOfflineMapWithLocalBasemap.java

```diff
@@ -0,0 +1,331 @@
+
+      // create a map with the portal item
+      map = new ArcGISMap(portalItem);
+      addMapLoadedListener();
+      // set the map to the map view
+      mapView.setMap(map);
+      // update the download area box whenever the viewpoint changes
+      mapView.addViewpointChangedListener(viewpointChangedEvent -> updateDownloadArea());
+
+      // create a progress bar to show download progress
+      createProgressIndicator();
+
+      // when the take map offline button is clicked, start the offline map task job
+      offlineMapButton.setOnAction(e -> {
+        try {
+          offlineMapButton.setVisible(false);
+          progressBar.setVisible(true);
+
+          // specify the extent, min scale, and max scale as parameters
+          double minScale = mapView.getMapScale();
+          double maxScale = map.getMaxScale();
+          // minScale must always be larger than maxScale
+          if (minScale <= maxScale) {
+            minScale = maxScale + 1;
+          }
+
+          // create an offline map task with the map
+          task = new OfflineMapTask(map);
+          // create default generate offline map parameters
+          ListenableFuture<GenerateOfflineMapParameters> generateOfflineMapParametersListenableFuture = task.createDefaultGenerateOfflineMapParametersAsync(downloadArea.getGeometry(), minScale, maxScale);
+          // get the offline map parameters from the offline map task
+          generateOfflineMapParameters = generateOfflineMapParametersListenableFuture.get();
+          generateOfflineMapParametersListenableFuture.addDoneListener(() -> {
+
+            // define the directory in which the downloaded .tpk file relevant to this offline map is saved
+            downloadedBasemapSavedLocation = new File("./samples-data/naperville/").getAbsolutePath();
+            // get the name of the basemap file from the offline map as supplied by the map's author (in this instance naperville_imagery.tpk)
+            String referenceBasemapFileName = generateOfflineMapParameters.getReferenceBasemapFilename();
+
+            // if the basemap file name isn't empty, search for that file name within local computer
+            if (!referenceBasemapFileName.isEmpty()) {
+              // search for the given file name within the samples-data directory
+              String localBasemapFileString = FilenameUtils.concat(downloadedBasemapSavedLocation, referenceBasemapFileName);
+              localBasemapFile = new File(localBasemapFileString);
+
+              if (localBasemapFile.exists()) {
+                // get the file's directory
+                localBasemapDirectory = localBasemapFile.getParent();
+
+                try {
+                  // open a new dialog box to alert user to the existence of the .tpk saved already on their machine
+                  chooseWhichBasemapToLoad();
+
+                } catch (IOException e1) {
+                  e1.printStackTrace();
```

Comment:
Only the top most try/catch prints the stack trace (because if an exception bubbles up that far, we likely don't know of it's cause well enough to communicate that in an alert). All other exception handling should show an alert to the user.